src/django_components/template_loader.py

```diff
@@ -56,18 +78,19 @@ def get_dirs(self) -> List[Path]:
             for app_config in apps.get_app_configs():
                 if not app_config.path:
                     continue
-                if not Path(component_dir).is_dir():
-                    continue
 
-                if path_is_relative_to(component_dir, app_config.path):
+                # NOTE: This applies only to absolute paths
+                if path_is_relative_to(component_dir, app_config.path) and Path(component_dir).is_dir():
                     curr_directories.add(Path(component_dir).resolve())
 
+            # NOTE: This applies only to relative paths
             if hasattr(settings, "BASE_DIR"):
-                path = (Path(settings.BASE_DIR) / component_dir).resolve()
+                path: Path = (Path(settings.BASE_DIR) / component_dir).resolve()
                 if path.is_dir():
                     curr_directories.add(path)
 
             # Add the directory that holds the settings file
+            # NOTE: This applies only to relative paths
             if settings.SETTINGS_MODULE:
```

Comment:
Same issue is here as I mention in https://github.com/EmilStenstrom/django-components/pull/533/files#r1649603172. Basically we treat the dirs in `STATICFILES_DIRS` as relative altho Django docs ask for absolute paths.

At first I didn't understand why we resolve relative to `SETTINGS_MODULE`, but turns out that was the initial implementation in https://github.com/EmilStenstrom/django-components/blob/58c6e476f1da7a227512f360946fc8f4f88796eb/django_components/template_loader.py. And then, couple of months ago from now, it was branched out in https://github.com/EmilStenstrom/django-components/blob/567c8ccc3d5747bc01a0a4b83c6d54f3880c4c7a/django_components/template_loader.py to fix https://github.com/EmilStenstrom/django-components/issues/290.

Correct me if wrong, but I think the code branch can be summarized as "Look for `component_dir` as relative subpath in settings dir (e.g. `my_app/settings/`), the app where settings are set (e.g. `my_app`), and the root". 

If that's right, we're practically trying to get to the same place as we do in the [code branch above](https://github.com/EmilStenstrom/django-components/pull/533/files#diff-756de4c35ec676b6ad61e551979238b310a7806c7c612add84aaec6f376c1b16R87) with `settings.BASE_DIR`. So we could just use `BASE_DIR` instead of relying on `SETTINGS_MODULE`.

And as I mention in [the comment above](https://github.com/EmilStenstrom/django-components/pull/533/files#r1649603172) on `BASE_DIR`, I think that it's simple enough to use to let users specify the full path via `BASE_DIR` as:
```py
Path(settings.BASE_DIR) / component_dir).resolve()
```

So personally I would remove this code block for `settings.SETTINGS_MODULE`, and tell users to replace that with something like:
```py
Path(settings.BASE_DIR) / "app_with_settings" / "settings").resolve()
```

Is there maybe something I'm missing about `settings.SETTINGS_MODULE`?

Reply:
UPDATE: I've refactored the Loader to use `BASE_DIR` instead of `SETTINGS_MODULE`